Recursor no longer compiles with el6 boost

### Short description

the current master version of the recursor no longer compiles with el6 boost. 

```
DEBUG:   CXX      opensslsigners.o
DEBUG:   CXX      pdns_recursor.o
DEBUG: pdns_recursor.cc:30:40: fatal error: boost/container/flat_set.hpp: No such file or directory
DEBUG:  #include <boost/container/flat_set.hpp>
DEBUG:                                         ^
DEBUG: compilation terminated.
DEBUG: make[2]: *** [pdns_recursor.o] Error 1
DEBUG: make[2]: *** Waiting for unfinished jobs....
DEBUG: make[2]: Leaving directory `/builddir/build/BUILD/pdns-recursor-0.0.2048gae1fd33'
DEBUG: make[1]: *** [all-recursive] Error 1
DEBUG: make[1]: Leaving directory `/builddir/build/BUILD/pdns-recursor-0.0.2048gae1fd33'
DEBUG: make: *** [all] Error 2
```

### Environment
<!-- Tell us about the environment -->
 - Operating system: EL6
 - Software version: ae1fd33
 - Software source: git

--- a/pdns/pdns_recursor.cc
+++ b/pdns/pdns_recursor.cc
@@ -26,8 +26,9 @@
 #include <netdb.h>
 #include <sys/stat.h>
 #include <unistd.h>
-
+#ifdef HAVE_BOOST_CONTAINER_FLAT_SET_HPP
 #include <boost/container/flat_set.hpp>
+#endif
 #include "ws-recursor.hh"
 #include <pthread.h>
 #include "recpacketcache.hh"
@@ -161,7 +162,11 @@ static bool g_gettagNeedsEDNSOptions{false};
 static time_t g_statisticsInterval;
 static bool g_useIncomingECS;
 std::atomic<uint32_t> g_maxCacheEntries, g_maxPacketCacheEntries;
+#ifdef HAVE_BOOST_CONTAINER_FLAT_SET_HPP
 static boost::container::flat_set<uint16_t> s_avoidUdpSourcePorts;
+#else
+static std::set<uint16_t> s_avoidUdpSourcePorts;
+#endif
 static uint16_t s_minUdpSourcePort;
 static uint16_t s_maxUdpSourcePort;
 

--- a/pdns/recursordist/configure.ac
+++ b/pdns/recursordist/configure.ac
@@ -90,6 +90,10 @@ AS_IF([test "x$PROTOBUF_LIBS" != "x" -a x"$PROTOC" != "x"],
 )
 
 BOOST_REQUIRE([$boost_required_version])
+
+# Check against flat_set header that requires boost >= 1.48
+BOOST_FIND_HEADER([boost/container/flat_set.hpp])
+
 PDNS_SELECT_CONTEXT_IMPL
 
 PDNS_ENABLE_UNIT_TESTS

